src/libstd/panicking.rs

```diff
@@ -508,8 +510,16 @@ pub extern fn rust_begin_panic(msg: fmt::Arguments,
            reason = "used by the panic! macro",
            issue = "0")]
 #[inline(never)] #[cold]
-pub fn begin_panic_fmt(msg: &fmt::Arguments,
-                       file_line_col: &(&'static str, u32, u32)) -> ! {
+pub fn begin_panic_fmt(msg: &fmt::Arguments, file_line_col: &(&'static str, u32, u32)) -> ! {
+    rust_panic_fmt(msg, file_line_col, begin_panic_fmt as usize);
+}
+
+#[cold]
+#[inline(never)]
+#[cfg_attr(not(stage0), notail_when_called)]
```

Comment:
Could you add comment as well to functions in this crate for why this attribute is applied?